test/parallel/test-module-main-fail.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const child_process = require('child_process');
+const { promisify } = require('util');
+const execFile = promisify(child_process.execFile);
+
+common.crashOnUnhandledRejection();
+
+const entryPoints = ['iDoNotExist', 'iDoNotExist.js', 'iDoNotExist.mjs'];
+const flags = [[], ['--experimental-modules']];
+const node = process.argv[0];
+
+(async () => {
+  for (const args of flags) {
+    for (const entryPoint of entryPoints) {
+      try {
+        await execFile(node, args.concat(entryPoint));
```

Comment:
Could the test be simplified by using `execFileSync()` instead?

Reply:
But isn't using async-await cooler 😁 ?

Reply:
Haha I totally forgot there was a sync version! I was only thinking about how complicated it would be with callbacks xD.